fuzzy-search.js

```diff
@@ -36,10 +38,10 @@ function fuzzySearch(haystackList, needle, renderer, caseSensitive = false) {
 	};
 
 	// Store preciseMatch and fuzzyMatch separately, to avoid running sort on preciseMatch
-	let preciseMatch = [];
+	let preciseMatch = "";
 	let fuzzyMatch = [];
-
-	for (let i = 0; i < haystackList.length; i++) {
+	const hayStackLength = haystackList.length
+	for (let i = 0; i < hayStackLength; i++) {
```

Comment:
Remove ` haystackList.length ` from for loop's `condition expression`  and store it in  a constant `hayStackLength`, so that the `length` property of `haystackList` is called only once